# 🎬 Movie Ratings Analysis — Exploration Notebook

This notebook loads the raw `movies.csv` dataset and performs an
initial inspection: shape, columns, dtypes, summary statistics, and a
first look at data quality issues (missing values, duplicates).


In [4]:
import sys

sys.path.append("..")

import pandas as pd

from src.data_loader import load_data

pd.set_option("display.max_columns", None)

## Load the raw dataset

In [5]:
df = load_data("../data/movies.csv")
df.head()

,Film,Genre,Lead Studio,Audience Score %,Profitability,Rotten Tomatoes %,Worldwide Gross,Year
0,Endless Mirage,Action,Focus Features,73.0,4.89,43.0,"$115,567,316.00",2008
1,Last Symphony,Documentary,Warner Bros.,53.0,4.14,81.0,"$290,886,656.00",2019
2,Endless Wanderer,Sci-Fi,MGM,59.0,2.70,76.0,"$148,035,902.00",2014
3,Shattered Mirage,Horror,20th Century Studios,69.0,1.03,47.0,"$61,118,528.00",2022
4,Electric Garden,Romance,Lionsgate,77.0,1.02,5.0,"$486,011,064.00",2018


## Shape and columns

In [6]:
print("Shape:", df.shape)
print("Columns:", list(df.columns))

Shape: (226, 8)
Columns: ['Film', 'Genre', 'Lead Studio', 'Audience Score %', 'Profitability', 'Rotten Tomatoes %', 'Worldwide Gross', 'Year']


## Data types and non-null counts

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 226 entries, 0 to 225
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Film               226 non-null    object 
 1   Genre              226 non-null    object 
 2   Lead Studio        218 non-null    object 
 3   Audience Score %   218 non-null    float64
 4   Profitability      226 non-null    float64
 5   Rotten Tomatoes %  218 non-null    float64
 6   Worldwide Gross    226 non-null    object 
 7   Year               226 non-null    int64  
dtypes: float64(3), int64(1), object(4)
memory usage: 14.3+ KB


## Summary statistics

In [8]:
df.describe()

,Audience Score %,Profitability,Rotten Tomatoes %,Year
count,218.000000,226.000000,218.000000,226.000000
mean,64.743119,4.196018,57.724771,2014.853982
std,15.087703,4.477985,18.798653,5.223103
min,25.000000,0.190000,5.000000,2007.000000
25%,54.250000,1.552500,46.250000,2009.000000
50%,65.000000,2.855000,59.000000,2015.000000
75%,75.000000,4.897500,69.000000,2019.000000
max,99.000000,31.180000,99.000000,2023.000000


## Missing values and duplicates

In [9]:
print("Missing values per column:")
print(df.isna().sum())
print("\nDuplicate rows:", df.duplicated().sum())

Missing values per column:
Film                 0
Genre                0
Lead Studio          8
Audience Score %     8
Profitability        0
Rotten Tomatoes %    8
Worldwide Gross      0
Year                 0
dtype: int64

Duplicate rows: 6


## Findings

- 226 rows, 8 columns.
- `Lead Studio`, `Audience Score %`, and `Rotten Tomatoes %` contain missing values.
- `Worldwide Gross` is stored as a currency-formatted string, not numeric.
- A handful of duplicate rows exist.


